recipes/libssh/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir
+from conan.tools.microsoft import is_msvc_static_runtime, is_msvc
+import os
+
+
+required_conan_version = ">=1.54.0"
+
+
+class LibSSHRecipe(ConanFile):
+    name = "libssh"
+    license = "LGPL-2.1"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.libssh.org/"
+    description = "multiplatform C library implementing the SSHv2 protocol on client and server side"
+    topics = ("ssh", "shell", "ssh2", "connection")
+    package_type = "library"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "crypto_backend": ["openssl", "gcrypt"],
+        "with_symbol_versioning": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "crypto_backend": "openssl",
+        "with_symbol_versioning": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            del self.options.with_symbol_versioning
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_zlib:
+            self.requires("zlib/[>=1.2.11 <2]")
+        if self.options.crypto_backend =="openssl":
+            self.requires("openssl/[>=1.1 <4]")
+        elif self.options.crypto_backend == "gcrypt":
+            self.requires("libgcrypt/1.8.4")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CLIENT_TESTING"] = False
+        tc.variables["SERVER_TESTING"] = False
+        tc.variables["WITH_EXAMPLES"] = False
+        tc.variables["WITH_GCRYPT"] = self.options.crypto_backend == "gcrypt"
+        tc.variables["WITH_GSSAPI"] = False
+        tc.variables["WITH_MBEDTLS"] = False
```

Comment:
quick question here, did you try to use the mbedtls package here? https://conan.io/center/recipes/mbedtls?version=3.6.0

It's not a blocker, but I would recommend adding a TODO comment to improve it in the future.